02_TrackWeb/src/db.ts

```
@@ -0,0 +1,135 @@
+  }
+
+  db = await openDB<WebTimeTrackerDBSchema>(DB_NAME, DB_VERSION, {
+    upgrade(db: IDBPDatabase<WebTimeTrackerDBSchema>, oldVersion: number, newVersion: number, transaction: IDBPTransaction<WebTimeTrackerDBSchema, 'website_times'[], 'versionchange'>) {
+      switch (oldVersion) {
+        case 0:
+          // Migration from no database to version 1
+          db.createObjectStore(STORE_NAME, {
+            keyPath: 'normalizedUrl',
+          });
+          // You might want to create indexes here if needed for performance on queries other than keyPath
+          const store = transaction.objectStore(STORE_NAME);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)